File: poo/java/Estacionamento/Horista.java
/*
 *  Programa escrito com propósito didático apenas.
 *  @author Nathan Cruz <nathan.cruz@unifesp.br>
 *  @created 11/04/14
 *  R.A.: 86492
 */

public class Horista extends Veiculo{

    /* Atributos */
    private double valorHora;
    private int horasEstacionado;

    /* Construtor */
    public Horista(String placa, String cor, String modelo,
                   double valorHora, int horasEstacionado){
        super.setPlaca(placa);
        super.setCor(cor);
        super.setModelo(modelo);
        this.setValorHora(valorHora);
        this.setHorasEstacionado(horasEstacionado);
    }

    /* Getters */
    public double getValorHora(){
        return this.valorHora;
    }
    public int getHorasEstacionado(){
        return this.horasEstacionado;
    }

    /* Setters */
    public void setValorHora(double valorHora){
        this.valorHora = valorHora;
    }
    public void setHorasEstacionado(int horasEstacionado){
        this.horasEstacionado = horasEstacionado;
    }

    /* Declarado na super-classe */
    public double PrecoPagar(){
        double preco =  this.getValorHora() * this.getHorasEstacionado();
        return preco;
    }
}

File: poo/python/Estacionamento/Horista.py
#!/usr/bin/env python
#-*- coding: utf-8 -*-

"""
@author Nathan Cruz <nathan.cruz@unifesp.br>
@created 12/04/14
"""
from __future__ import unicode_literals
from __future__ import print_function

from Veiculo import Veiculo

class Horista(Veiculo):

    # Atributos #
    __valorHora = None
    __horasEstacionado = None

    # Construtor #
    def __init__(self, placa, cor, modelo, valorHora, horasEstacionado):
        super(Horista, self).setPlaca(placa)
        super(Horista, self).setCor(cor)
        super(Horista, self).setModelo(modelo)
        self.setValorHora(valorHora)
        self.setHorasEstacionado(horasEstacionado)

    # Getters #
    def getValorHora(self):
        return self.__valorHora
    def getHorasEstacionado(self):
        return self.__horasEstacionado

    # Setters #
    def setValorHora(self, valorHora):
        self.__valorHora = valorHora
    def setHorasEstacionado(self, horasEstacionado):
        self.__horasEstacionado = horasEstacionado

    # Redefinida #
    def PrecoPagar(self):
        return (self.getValorHora()*self.getHorasEstacionado())
